lib/widget/switch.dart

```diff
@@ -0,0 +1,140 @@
+import 'package:flutter/widgets.dart';
+import 'package:flutter_neumorphic/NeumorphicBoxShape.dart';
+import 'package:flutter_neumorphic/flutter_neumorphic.dart';
+
+// Constants below are for easy scaling tests.
+// Can be removed later.
+const double SCALE = 1.0;
+const double TRACK_WIDTH = 90 * SCALE;
+const double TRACK_HEIGHT = 45 * SCALE;
+const double THUMB_WIDTH = 31.5 * SCALE;
+const double THUMB_HEIGHT = 31.5 * SCALE;
+const double THUMB_CONTAINER_WIDTH = 90 * SCALE;
+const double THUMB_CONTAINER_HEIGHT = 36 * SCALE;
+
+class NeumorphicSwitchStyle {
+  final double depth;
+  final Color activeTrackColor;
+  final Color inactiveTrackColor;
+  final Color activeThumbColor;
+  final Color inactiveThumbColor;
+
+  const NeumorphicSwitchStyle(
+      {this.depth = -3,
```

Comment:
remove this depth, if not, updating the theme.depth will have no impact on this widget